datasource_generic.go

```diff
@@ -0,0 +1,130 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+)
+
+type DataSourceGeneric struct {
+	Id     int64  `json:"id,omitempty"`
+	Name   string `json:"name"`
+	Type   string `json:"type"`
+	URL    string `json:"url"`
+	Access string `json:"access"`
+
+	Database string `json:"database,omitempty"`
+	User     string `json:"user,omitempty"`
+	// Deprecated in favor of secureJsonData.password
+	Password string `json:"password,omitempty"`
+
+	OrgId     int64 `json:"orgId,omitempty"`
+	IsDefault bool  `json:"isDefault"`
+
+	BasicAuth     bool   `json:"basicAuth"`
+	BasicAuthUser string `json:"basicAuthUser,omitempty"`
+	// Deprecated in favor of secureJsonData.basicAuthPassword
+	BasicAuthPassword string `json:"basicAuthPassword,omitempty"`
+
+	JSONData       JsonData `json:"jsonData,omitempty"`
+	SecureJSONData JsonData `json:"secureJsonData,omitempty"`
+}
+
+type JsonData map[string]interface{}
```

Comment:
Does this type have to be exported?
```suggestion
type JSONData map[string]interface{}
```

Reply:
This is copy/paste of https://github.com/jwierzbo/grafana-api-golang-client/blob/4cd08a254b133d2ecabaa6c052008646f6b25524/datasource.go#L30

Let's stay consistent